packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts

```diff
@@ -65,6 +70,8 @@ import { arrayFromCollection, isGlobalPatchingSkipped } from '../shared/utils';
 import { getNodeOwnerKey, isNodeShadowed } from '../faux-shadow/node';
 import { assignedSlotGetterPatched } from './slot';
 
+const DISABLE_ELEMENT_PATCH = !ENABLE_ELEMENT_PATCH;
```

Comment:
im not proud of this, but this negation only happens once, against every time that is used.

This is to match the proposed behavior from @ekashida :

```js
if (flag is disabled) {
  // dont patch
}

// Regular code path 
```
 
which i agree looks cleaner.

Reply:
This won't work because the transform for dynamic feature flags only kicks in if you use the imported binding directly in the if-test like `if (ENABLE_FOO) { ... }`. And FYI, `if (!ENABLE_FOO) { ... }` will not work.

Reply:
mm, im definitely missing something about the feature flags.

it is tree-shaking when compiling the engine and you set the flag to true or false even while set to null. you can check in `lwc/dist/synthetic-shadow` that is removing the code inside the if flag.

the engine, wire-service and synthetic-module are only compiled once to generate the distribution and this feature, is a runtime flag (which at this point is not because of the tree shaking).

Reply:
@ekashida i believe `@lwc/synthetic-shadow` module wasn't using in the build process the feature flags. i addressed that in this commit: https://github.com/salesforce/lwc/pull/1550/commits/2d681081436335733f00d9d17c0a48592f9f7f60

Also did this: https://github.com/salesforce/lwc/pull/1550/commits/191b03373c55a3c6626b3e0388703b272635ea1c in order to resolve your previous comment. We can do it assuming that setting the flags is prior this code is executed.

Once we agree on the names for the flags, we can revisit this.